src/neapolitan/views.py

```diff
@@ -151,6 +153,37 @@ def get_queryset(self):
         )
         raise ImproperlyConfigured(msg % self.__class__.__name__)
 
+    def get_fields(self):
+        # Construct the method name based on the current role.
+        # For example, if self.role is Role.LIST, method_name will be 'list_fields'.
+        method_name = f"{self.role}_fields"
```

Comment:
This doesn't quite work because of for `self.role` because `get_firelds()` gets called by `get_form_class()` which is always either `create_fields` or `update_fields`. I'm still thinking this over.